source/material_model/dynamic_friction.cc

```diff
@@ -0,0 +1,444 @@
+          thermal_conductivities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Thermal conductivities"))),
+                                                                           n_fields,
+                                                                           "Thermal conductivities");
+          thermal_expansivities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Thermal expansivities"))),
+                                                                          n_fields,
+                                                                          "Thermal expansivities");
+          specific_heats = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Specific heats"))),
+                                                                   n_fields,
+                                                                   "Specific heats");
+          prm.enter_subsection("Viscosities");
+          {
+            minimum_viscosity  = prm.get_double ("Minimum viscosity");
+            maximum_viscosity  = prm.get_double ("Maximum viscosity");
+            reference_strain_rate    = prm.get_double ("Reference strain rate");
+
+            mu_s = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Coefficients of static friction"))),
+                                                                  n_fields,
+                                                                  "Coefficients of static friction");
+            mu_d = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Coefficients of dynamic friction"))),
+                                                                  n_fields,
+                                                                  "Coefficients of dynamic friction");
+            cohesions = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Cohesions"))),
+                                                                  n_fields,
+                                                                  "Cohesions"); 
+            background_viscosities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Background Viscosities"))),
+                                                                  n_fields,
+                                                                  "Background Viscosities"); 
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::compositional_fields;
```

Comment:
This material model has a viscosity that is also dependent on the strain rate, has it not?